linera-chain/src/data_types.rs

```diff
@@ -171,23 +171,24 @@ pub struct OutgoingEffect {
     pub effect: Effect,
 }
 
-/// The type of a value: whether it's a validated block in a particular round, or already
-/// confirmed.
-#[derive(Debug, PartialEq, Eq, Hash, Clone, Copy, Serialize, Deserialize)]
-pub enum ValueKind {
-    /// The block was validated but confirmation will require additional steps.
-    ValidatedBlock { round: RoundNumber },
-    /// The block is validated and confirmed (i.e. ready to be published).
-    ConfirmedBlock,
+/// A block, together with the effects and the state hash resulting from its execution.
+#[derive(Debug, PartialEq, Eq, Hash, Clone, Serialize, Deserialize)]
+pub struct ExecutedBlock {
+    pub block: Block,
+    pub effects: Vec<OutgoingEffect>,
+    pub state_hash: CryptoHash,
 }
 
 /// A statement to be certified by the validators.
 #[derive(Debug, PartialEq, Eq, Hash, Clone, Deserialize, Serialize)]
-pub struct Value {
-    pub block: Block,
-    pub effects: Vec<OutgoingEffect>,
-    pub state_hash: CryptoHash,
-    pub kind: ValueKind,
+pub enum Value {
+    ValidatedBlock {
+        executed: ExecutedBlock,
```

Comment:
I believe the getters should be able to avoid `block.block` if needed.

Reply:
I renamed `executed` to `executed_block`, and avoided most (not all) cases of `executed_block.block`, but using pattern matching, not a getter, because we'll add variants without a block later. And I think the only place in the code where we legitimately want something out of the block, regardless of whether it's a confirmed or only validated one, is when we get the bytecode locations of its dependencies.